pkg/universe.dagger.io/go/test.cue

```diff
@@ -2,13 +2,32 @@ package go
 
 // Test a go package
 #Test: {
-	// Package to test
-	package: *"." | string
+	// DEPRECATED: use packages instead
+	package: string | *null
+
+	// Packages test
+	packages: [...string]
 
 	#Container & {
 		command: {
+			//FIXME: find a better workaround with disjunction
+			//FIXME: factor with the part from test.cue
+			_packages: [...string]
+			if package == null && len(packages) == 0 {
+				_packages: ["."]
+			}
+			if package != null && len(packages) == 0 {
+				_packages: [package]
+			}
+			if package == null && len(packages) > 0 {
+				_packages: packages
+			}
+			if package != null && len(packages) > 0 {
+				_packages: [package] + packages
+			}
```

Comment:
I think this would be simpler:
```cue
package: *"." | string     // notice previous default preserved
packages: *[package] | [...string]   // previous behavior preservered as a default, allows overrides

#Container & {
    command:
        args: packages
        ...
}
```

This would mean that you wouldn't be able to use both at the same time, which is fine by me. The intent is to not break current configs, so if you're setting both `package` and `packages` you're updating to the new field, in which case I think it's ok to not make `package` work.

Ideally user should be warned if using both, but that will happen when we build the deprecation warnings. If we allow both to be used at the same time it also sets a precedent that that is a working config.

Reply:
This was the reason this took so much time to fix.
And I thought about it enough to believe it is the right way.

## My suggestion:  `package` ∪ `packages` 

## Pros

- allow normal functioning even when mistakes are made by users (they didn't know `package` was being deprecated)
- function even if somewhere else in the package, somebody modify the `package(s)` fields

## Cons

- allow people to make mistake without thinking twice
- worse performances
- worse code/duplication

## Your suggestion: `package` Δ `packages`

## Pros

- less package code, cleaner
- better performance

## Cons

- user that make mistakes won't understand why they set `package` or `packages`, and it is ignored, unless they read our CUE package code. We might get more support.

Reply:
Can we have more opinions on which direction to take for this PR?

basically, my PR makes `package` + `packages` merge into a group of packages to test. But the code needed some if, otherwise we fell into disjunction land. So less performant.

@helderco thinks we should just have 1 option active at runtime, with preferences on `packages`, so if `package` and `packages` is set, test would ignore the former, and would run only on `packages`. User might be confused if they configured `package` and it is not tested at all.

Since it's the oldest opened PR, we could finally close it.

cc: @helderco @TomChv @aluzzardi @samalba @shykes

Reply:
This looks straightforward to me:

```cue
package: *"." | string // notice previous default preserved packages: *[package] | [...string] // previous behavior preservered as a default, allows overrides
packages: *[package] | [...string] // previous behavior preservered as a default, allows overrides
```

Showing a deprecation warning at the top that does not get hidden by the default tty logger sounds like a nice way of eventually removing `package`. I think that it's OK for users that declare both to ask for support. I assume that there will not be more than a handful that do it before we remove `package`.

Reply:
Yes, I prefer the deprecation option and then perhaps prioritize #2251 so that we can actually warn users about it.

```cue
// DEPRECATED: Use `packages` instead
package: *"." | string
packages: *[package] | [...string]
```

With that change, it'll become:

```cue
package: *"." | string @deprecated("use `packages` instead")
packages: *[package] | [...string]
```

And then we can detect on load if that field is used and trigger a warning with that message.

In any case, I'm also ok to merge as is and simplify/deprecate later when we have the warnings support. If so, please add a TODO for this.

Reply:
How do we resolve this one @dolanor?

Reply:
I like this deprecation notice. As talked with @gerhard, I'll already fix this PR with this soon to come update (it doesn't impact the plan, still works).
Also, we decided with @gerhard that finally, the package+packages use case might be too niche to bother and it might not create too many problems in the future (which is what I feared).
Also, the fix is **really** easy, `s/package/packages/`.